test/RollingFileWriteStream-test.js

```diff
@@ -16,7 +15,7 @@ const RollingFileWriteStream = proxyquire('../lib/RollingFileWriteStream', {
 
 function generateTestFile(fileName) {
   const dirName = path.join(__dirname, 'tmp_' + Math.floor(Math.random() * new Date()));
-  fileName = fileName || 'ignore.log';
+  fileName = fileName || (Math.floor(Math.random() * Math.floor(2)) ? 'withExtension.log' : 'noExtension');
```

Comment:
I wouldn't use random for tests - you can never be sure that it will test the things you want to test, and you might get weird cases where a test will pass the first time you run it, but then fail the second time. Better to put explicit test cases in that will run every time.